plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/CustomizeStore.php

```diff
@@ -123,6 +193,184 @@ public function get_action_url() {
+			'is_in_stock'         => true,
+			'is_on_backorder'     => false,
+			'low_stock_remaining' => null,
+			'sold_individually'   => false,
+			'extensions'          => array(),
+		);
+	}
+
+	/**
+	 * Adds the rendered title to the product response.
+	 *
+	 * This method sets the 'rendered' title of a product to be the same as its 'raw' title
+	 * in the REST API response.
+	 *
+	 * @param WP_REST_Response $response The response object.
+	 * @param WP_Post          $post     The original post object.
+	 * @param WP_REST_Request  $request  The request object.
+	 * @return WP_REST_Response The modified response object.
+	 */
+	public function add_product_rendered_title( $response, $post, $request ) {
+		if ( ! empty( $response->data['title']['rendered'] ) || 'product' !== $post->post_type ) {
+			return $response;
+		}
+
+		$response->data['title']['rendered'] = $response->data['title']['raw'];
+
+		return $response;
+	}
+
+	/**
+	 * Adds sample products to the query result if no products are found.
+	 *
+	 * This function is used to populate the product list with sample data when
+	 * the store has no real products and a REST API request is made.
+	 *
+	 * @param array    $posts An array of post objects.
+	 * @param WP_Query $query The WP_Query instance.
+	 * @return array An array of post objects, potentially including sample products.
+	 */
+	public function add_sample_products_to_query( $posts, $query ) {
+		if ( 'product' !== $query->query['post_type'] || ! WC()->is_rest_api_request() || ! empty( $posts ) ) {
+			return $posts;
+		}
+
+		$sample_products = array();
+
+		foreach ( $this->sample_products as $index => $sample_product ) {
+			$sample_products[] = $this->generate_sample_product_post_object( $index );
+		}
+
+		return $sample_products;
+	}
+
+	/**
+	 * Handles requests for nonexistent products by generating sample product data.
+	 *
+	 * This method intercepts REST API requests for product details. If the requested
+	 * product doesn't exist and the product ID suggests it's a sample product
+	 * (ID representing a timestamp within the last 24 hours), it returns a
+	 * sample product object.
+	 *
+	 * @param mixed           $result  The current result, usually null for not found.
+	 * @param WP_REST_Server  $server  The REST server instance.
+	 * @param WP_REST_Request $request The current REST request.
+	 *
+	 * @return WP_REST_Response|mixed The sample product response if applicable, otherwise the original result.
+	 */
+	public function add_sample_product_data_to_query( $result, $server, $request ) {
+		if ( false !== strpos( $request->get_route(), '/wc/store/v1/products/' ) ) {
+			$product_id = (int) str_replace( '/wc/store/v1/products/', '', $request->get_route() );
+			$product    = wc_get_product( $product_id );
+
+			// Sample products use `time()` as the ID. This way we can detect that
+			// it was a sample product returned 24 hours or less ago.
+			if ( ! $product && $product_id + 86400000 > time() * 1000 ) {
```

Comment:
Since product are dummy data we have full control over the structure of the object, why no adding a specific property to the product so we can know it is dummy, instead of doing this hack. Maybe we can add a meta called is_dummy or just a prop called similar, perhaps we could use another status. If we need to know when the product was created we could still use the `post_date` prop. This way, this logic will be more natural.

Reply:
I refactored this logic a bit here: 2b24c07a8e5d98eadb999449b7e71a83d62f3ed6.